src/shims/foreign_items.rs

```diff
@@ -504,6 +508,7 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
 
             // Architecture-specific shims
             "llvm.x86.sse2.pause" if this.tcx.sess.target.arch == "x86" || this.tcx.sess.target.arch == "x86_64" => {
+                check_abi(abi, Abi::C)?;
                 let &[] = check_arg_count(args)?;
                 this.yield_active_thread();
```

Comment:
I just added another architecture-specific function here; can you rebase and also add the `check_abi` here?